app/src/main/java/com/example/androidseminar/SignInActivity.kt

```diff
@@ -0,0 +1,109 @@
+package com.example.androidseminar
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.content.SharedPreferences
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import com.example.androidseminar.databinding.ActivitySignInBinding
+
+class SignInActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivitySignInBinding
+    private val test = "log"
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignInBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        Log.d(test, "SignInActivity - onCreate")
+
+        buttonClickEvent()
+    }
+
+    private fun buttonClickEvent() {
+        val userId = binding.editId.text
+        val userPw = binding.editPw.text
+
+        val prefs : SharedPreferences = getSharedPreferences("Login", Context.MODE_PRIVATE)
+
+        binding.btnLogin.setOnClickListener {
+            val name = prefs.getString("name", "")
+            val id = prefs.getString("id", "")
+            val pw = prefs.getString("pw", "")
+            if (userId.isNullOrBlank() || userPw.isNullOrBlank()) {
+                Toast.makeText(this, "아이디/비밀번호를 확인해주세요!", Toast.LENGTH_SHORT).show()
+            } else if (userId.toString() == id && userPw.toString() == pw) {
+                Toast.makeText(this, "로그인 성공", Toast.LENGTH_SHORT).show()
+                val intent = Intent(this, HomeActivity::class.java)
+                intent.putExtra("userId",id)
+                intent.putExtra("userName",name)
+                startActivity(intent)
+            } else {
+                Toast.makeText(this, "아이디/비밀번호가 일치하지 않습니다.", Toast.LENGTH_SHORT).show()
+            }
+        }
+        binding.tvSignUp.setOnClickListener {
+            val intent = Intent(this, SignUpActivity::class.java)
+            startActivityForResult(intent, 100)
```

Comment:
startActivityForResult()를 사용하면 Result Code가 필요한데 이렇게 직접 100을 하드코딩해서 넣는 것보다 따로 상수로 선언해서 넣어주는걸 추천해:)   

이런식으로!?!?


```suggestion
      binding.tv.setOnClickListener{
       startActivityForResult(intent, SIGNUP_REQUEST_CODE)
      }
            
companion object { 
      private const val SIGNUP_REQUEST_CODE = 100 
}
```

Reply:
알면서도 하드코딩을 하게 되네,, 이참에 습관부터 바꿔볼게 :)